### PR TITLE
f-account-info@v0.8.0 - Add the Vuex store (get data only)

### PR DESCRIPTION
**Pull request by oliversweb**
### Added
- Vuex store - get, map & store consumer details/address data
---
This is just the getting of the data .... the saving of the data will follow once we have the spec.

### CODE REVIEW

File: packages/components/pages/f-account-info/src/services/mapping/consumer.mapper.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+/**
+ * Validates the parameter passed is an Array then finds the default consumers address
+ * then maps the raw api response to the view model.
+ * @param {object} addresses - The consumers addresses (i.e. response.data).
+ * @returns {object} A view model that can be used to presents the component.
+ */
+export const mapToConsumerAddress = addresses => {
+    let address;
+    if (Array.isArray(addresses)) {
+        address = addresses.find(e => e.IsDefault);
```
Comment (jamieomaguire) [inline packages/components/pages/f-account-info/src/services/mapping/consumer.mapper.js:28]: Do you need a chained `?.` in the event of e being null?

File: packages/components/pages/f-account-info/src/store/accountInfo.module.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+            const responses = await Promise.all([getDetailsTask, getAddressTask]);
+
+            const details = mapToConsumerDetails(responses[0]?.data);
+            const address = mapToConsumerAddress(responses[1]?.data);
+
+            commit(UPDATE_CONSUMER_DETAILS, { details, address });
+        },
+
+        async editConsumerDetails ({ commit, state }, { field, value }) {
```
Comment (jamieomaguire) [inline packages/components/pages/f-account-info/src/store/accountInfo.module.js]: is async needed here if nothing is being awaited?

Comment (oliversweb) [reply packages/components/pages/f-account-info/src/store/accountInfo.module.js]: Good spot 👍

File: packages/components/pages/f-account-info/src/services/mapping/consumer.mapper.js (hunk 1 above)
Comment (oliversweb) [reply packages/components/pages/f-account-info/src/services/mapping/consumer.mapper.js:28]: Added test to prove `e` is not evaluated in the case of no elements 👍

**Review by jamieomaguire — APPROVED** [review]

**Review by AndyMayJE — APPROVED** [review]

**Pull request merged**